From 962015a52206d6b004ff59aa123afee398f38e8b Mon Sep 17 00:00:00 2001
From: Vasiliy Doylov <nekocwd@mainlining.org>
Date: Mon, 10 Nov 2025 11:43:45 +0300
Subject: [PATCH 114/212] arm64: dts: qcom: sdm845-oneplus: add rear camera
 actuators

This commit enables sub and main rear camera actuators

Signed-off-by: Vasiliy Doylov <nekocwd@mainlining.org>
---
 .../boot/dts/qcom/sdm845-oneplus-common.dtsi  | 24 ++++++++++++++++---
 1 file changed, 21 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi b/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
index 2a749b0372b7..75ed3974de51 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
@@ -541,6 +541,13 @@ cam_vio_1p8:
 		vreg_lvs1_1p8: lvs1 {
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1800000>;
+
+			/*
+			 * Supplies the 1.8V side of the CCI I2C level
+			 * shifter; keep enabled until a generic
+			 * bus-supply property exists.
+			 */
+			regulator-always-on;
 		};
 
 	};
@@ -669,6 +676,8 @@ rear_cam: camera@1a {
 		orientation = <1>; /* Rear camera */
 		rotation = <270>;
 
+		lens-focus = <&rear_cam_actuator>;
+
 		port {
 			imx519_ep: endpoint {
 				bus-type = <MEDIA_BUS_TYPE_CSI2_CPHY>;
@@ -679,7 +688,11 @@ imx519_ep: endpoint {
 		};
 	};
 
-	/* rear onnn,lc898217xc @72 */
+	rear_cam_actuator: actuator@72 {
+		compatible = "onnn,lc898217xc";
+		reg = <0x72>;
+		vdd-supply = <&main_cam_vaf>;
+	};
 };
 
 &cci_i2c1 {
@@ -705,6 +718,8 @@ sub_cam: camera@10 {
 		orientation = <1>; /* Rear camera */
 		rotation = <270>;
 
+		lens-focus = <&sub_cam_actuator>;
+
 		port {
 			imx376k_ep: endpoint {
 				remote-endpoint = <&csiphy1_ep>;
@@ -712,10 +727,13 @@ imx376k_ep: endpoint {
 				link-frequencies = /bits/ 64 <500000000>;
 			};
 		};
-
 	};
 
-	/* onnn,lc898217xc @74 */
+	sub_cam_actuator: actuator@74 {
+		compatible = "onnn,lc898217xc";
+		reg = <0x74>;
+		vdd-supply = <&sub_cam_vaf>;
+	};
 };
 
 &cdsp_pas {
-- 
2.55.0

